packages/date-picker/src/vaadin-date-picker-helper.js

```diff
@@ -103,3 +103,28 @@ export function extractDateParts(date) {
     year: date.getFullYear(),
   };
 }
+
+/**
+ * Calculate the year of the date based on the provided reference date
```

Comment:
Let's extend the description to clarify that this function gets a two-digit year and returns a full year.